src/components/Layout/Layout.component.jsx

```diff
@@ -1,11 +1,19 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import NavBar from './NavBar.component';
+import { useGlobal } from '../../providers/GlobalContext/GlobalContext';
 
-// import './Layout.styles.css';
+import './Layout.styles.css';
 
 function Layout({ children }) {
+  const globalContext = useGlobal();
+  const [theme, setTheme] = useState('lightTheme');
+
+  useEffect(() => {
+    setTheme(globalContext.themeState);
+  }, [globalContext.themeState]);
```

Comment:
Whenever the context changes it will trigger a rerender on the components that use it.  Hence, you could remove the need of `useState`, and use the theme directly from the globalContext as:

```javascript
const { themeState } = globalContext;
```